### ISSUE TITLE
Add OpenTelemetry support

### ISSUE THREAD

**Issue by lahma** [quartznet/quartznet#867]
Umbrella to gather ideas and requirements.

**Comment by jbogard** [comment]
Think of this as two separate things - adding activity and diagnostics, and tracing/telemetry. It's straightforward to add activities and diagnostics, but tracing/telemetry assumes you have someone *listening* to those diagnostics.

If you look at my [NServiceBus](https://github.com/jbogard/NServiceBus.Extensions.Diagnostics) and [MongoDb](https://github.com/jbogard/MongoDB.Driver.Core.Extensions.DiagnosticSources) diagnostics repositories, those have two projects. One does the activity stuff, the other the telemetry stuff.

MongoDB is useful in particular because its operations don't require any kind of network correlation/propagation. It just starts activities and raises events via diagnostics listeners.

**Comment by lahma** [comment]
Added `DiagnosticListener` to Quartz's job run shell that notifies of job executions. Created separate project `Quartz.OpenTelemetry.Instrumentation` which follows patterns in sample integration projects they have.

@jbogard just a heads up, they've named things like Adapter -> Instrumentation as spec has changed, the latest betas in NuGet required some rework, I also used the naming they were using (in the new integration project).
